### DISCUSSION TITLE
How to reapeat last action ?

### DISCUSSION THREAD

**Discussion by ftwater** [JetBrains/ideavim#596] [Q&A]
I have write `nmap gmd <action>(MethodDown)`, when I press `gmd` it works ,now I need to keep on goinig to next method. how to do this just like use `.` in vim?

**Comment by ljchen1** [comment]
I doubt the `.` would work in VIM for your case because the "." command repeats the last change made in normal mode, but not for the motion command.
The `gmd` is a motion so the `.` won't repeat it. (see [here](https://stackoverflow.com/q/26553239/21033369) for similar cases).

Also, IdeaVim supports [m and ]m to move to the start of the previous/next method and [M and ]M to move to the end of the previous/next method and you could use it. (though it is not the exactly same as `<action>(MethodDown)`)

**Reply by miranquil** [reply]
Nah only `.` is not enough. I've encountered a situation today. I'm updating a method in my code by adding a new parameter and then I need to update all usages. So I need to execute `:action GoToNextError` repeatedly. Though I can nnoremap this action to a combination(and sure it is), I found that if an action like `:action RepeatLastIdeAction` would be better

**Comment by ManuelTS** [comment]
Vim macros work flawlessly for this.
